### ISSUE TITLE
reverse proxy for rpc?

### ISSUE THREAD

**Issue by Roki100** [monero-project/monero#8362]
Hello! I started my node few days ago and i am using nginx for my node now to connect to rpc with wallets, but i have noticed that monerod is not aware of this at all and when i enable log level to 1 each http request is actually an ip of nginx's docker container, is there a way to get monerod to be "aware" of the reverse proxy and recognize real ips via some real ip header? like `--reverse-proxy` or something?

**Comment by plowsof** [comment]
Do you have monerod running in a docker container? if so you can force the docker container to have a static ip [see here](https://github.com/plowsof/flipstarter-waas-wip/blob/32cd2628a4285437eaf698e8ee52b77e948e79ef/docker-compose.yml#L13) note: this is considered bad practice, but it works for me.

**Comment by Roki100** [comment]
> Do you have monerod running in a docker container? if so you can force the docker container to have a static ip [see here](https://github.com/plowsof/flipstarter-waas-wip/blob/32cd2628a4285437eaf698e8ee52b77e948e79ef/docker-compose.yml#L13) note: this is considered bad practice, but it works for me. 
> and then in nginx something like:
> ```
>   location /something {
>     proxy_pass http://172.20.111.2:8000;
>   }
> ```

yes it is running in docker container and i have already set up networking properly, proxy_pass https://monerod:18089;
but that is not what i asked though, monerod is not aware of being behind a proxy and it doesnt recognize real request ips, so i guess its a bad thing for any sort of anti abuse mechanisms

**Comment by plowsof** [comment]
sounds like a docker container / networking issue rather than monerod problem? It reminds me of running a tor node where all ip's are 'localhost', is there a specific issue that this causes? or are you worried about something potentially happening?

**Comment by Roki100** [comment]
> sounds like a docker container / networking issue rather than monerod problem? It reminds me of running a tor hidden service* where all ip's are 'localhost', is there a specific issue that this causes? or are you worried about something potentially happening?

no its not docker issue, monerod simply does not respect any real ip/forwarded for header
